packages/cordova-plugin/android/OSFileTransferPlugin.kt

```diff
@@ -0,0 +1,326 @@
+                }
+                listeners.clear()
+                true
+            }
+            else -> false
+        }
+    }
+
+    private fun parseDownloadOptions(args: JSONArray): OSFileTransferDownloadOptions? {
+        return try {
+            val optionsObject = args.getJSONObject(0)
+            val url = optionsObject.optString("url")
+            val path = optionsObject.optString("path")
+            
+            if (url.isBlank() || path.isBlank()) {
+                return null
+            }
+            
+            OSFileTransferDownloadOptions(
+                url = url,
+                path = path,
+                progress = optionsObject.optBoolean("progress", false),
+                httpOptions = parseHttpOptions(optionsObject, "GET")
+            )
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    private fun parseUploadOptions(args: JSONArray): OSFileTransferUploadOptions? {
+        return try {
+            val optionsObject = args.getJSONObject(0)
+            val url = optionsObject.optString("url")
+            val path = optionsObject.optString("path")
+            
+            if (url.isBlank() || path.isBlank()) {
+                return null
+            }
+            
+            OSFileTransferUploadOptions(
+                url = url,
+                path = path,
+                progress = optionsObject.optBoolean("progress", false),
+                mimeType = optionsObject.optString("mimeType"),
+                fileKey = optionsObject.optString("fileKey", "file"),
+                chunkedMode = optionsObject.optBoolean("chunkedMode", false),
+                httpOptions = parseHttpOptions(optionsObject, "POST")
+            )
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    private fun parseHttpOptions(options: JSONObject, defaultMethod: String): IONFLTRTransferHttpOptions {
+        val headers = options.optJSONObject("headers") ?: JSONObject()
+        val params = options.optJSONObject("params") ?: JSONObject()
+        
+        return IONFLTRTransferHttpOptions(
+            method = options.optString("method", defaultMethod),
+            headers = headers.toMap(),
+            params = params.toParamsMap(),
+            shouldEncodeUrlParams = options.optBoolean("shouldEncodeUrlParams", true),
+            readTimeout = options.optInt("readTimeout", 60000),
+            connectTimeout = options.optInt("connectTimeout", 60000),
+            disableRedirects = options.optBoolean("disableRedirects", false)
+        )
+    }
+
+    private fun JSONObject.toMap(): Map<String, String> {
+        val map = mutableMapOf<String, String>()
+        this.keys().forEach { key ->
+            map[key] = this.optString(key, "")
+        }
+        return map
+    }
+
+    private fun JSONObject.toParamsMap(): Map<String, Array<String>> {
```

Comment:
Same as in the other PR, not super important, but this could be refactored to:

```
private fun JSONObject.toParamsMap(): Map<String, Array<String>> {
    val map = mutableMapOf<String, Array<String>>()
    this.keys().forEach { key ->
        when (val value = this.opt(key)) {
            is String -> map[key] = arrayOf(value)
            is JSONArray -> {
                val values = (0 until value.length())
                    .mapNotNull { value.optString(it, null) }
                    .toTypedArray()
                if (values.isNotEmpty()) {
                    map[key] = values
                }
            }
        }
    }
    return map
}
```